### PR TITLE
Use bytes to represent AMQP property map

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Using bytes will maintain the type fidelity that the SDK has for the ApplicationProperties.

### CODE REVIEW

File: sdk/core/Azure.Core.Amqp/src/Shared/AmqpAnnotatedMessageConverter.cs [hunk 1]
```diff
@@ -329,7 +329,7 @@ public static AmqpAnnotatedMessage FromAmqpMessage(AmqpMessage source)
 
                 if (source.Properties.CorrelationId != null)
                 {
-                    message.Properties.CorrelationId = new AmqpMessageId(source.Properties.CorrelationId.ToString());
+                    message.Properties.CorrelationId = new AmqpMessageId(source.Properties.CorrelationId.ToString()!);
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core.Amqp/src/Shared/AmqpAnnotatedMessageConverter.cs:332]: This change was necessary due to the .NET 6 target used by the extension.

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/Grpc/ServiceBusGrpcSessionEndToEndTests.cs [hunk 2]
```diff
@@ -217,36 +221,12 @@ public class ServiceBusBindToSessionMessageAndAbandon
                     new AbandonRequest
                     {
                         Locktoken = message.LockToken,
-                        PropertiesToModify = {{ "key", new SettlementProperties { StringValue = "value"} }}
+                        PropertiesToModify = EncodeDictionary(new Dictionary<string, object> {{ "key", TimeSpan.FromSeconds(60)}})
```
Comment (JoshLove-msft) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/Grpc/ServiceBusGrpcSessionEndToEndTests.cs:224]: Add more test cases for different .NET primitives to verify roundtripping.

File: sdk/core/Azure.Core.Amqp/src/Shared/AmqpAnnotatedMessageConverter.cs [hunk 3]
```diff
@@ -659,7 +659,7 @@ private static bool TryGetValueBody(AmqpMessage source, out AmqpMessageBody? val
         ///
         /// <returns><c>true</c> if a message property value was able to be created; otherwise, <c>false</c>.</returns>
         ///
-        private static bool TryCreateNetPropertyFromAmqpProperty(
+        public static bool TryCreateNetPropertyFromAmqpProperty(
```
Comment (jsquire) [inline sdk/core/Azure.Core.Amqp/src/Shared/AmqpAnnotatedMessageConverter.cs:662]: These being bumped to `public` is just for testing?

File: sdk/core/Azure.Core.Amqp/src/Shared/AmqpAnnotatedMessageConverter.cs [hunk 4]
```diff
@@ -579,7 +579,7 @@ private static bool TryGetValueBody(AmqpMessage source, out AmqpMessageBody? val
         ///
         /// <returns><c>true</c> if an AMQP property value was able to be created; otherwise, <c>false</c>.</returns>
         ///
-        private static bool TryCreateAmqpPropertyValueFromNetProperty(
+        public static bool TryCreateAmqpPropertyValueFromNetProperty(
```
Comment (jsquire) [inline sdk/core/Azure.Core.Amqp/src/Shared/AmqpAnnotatedMessageConverter.cs:582]: nit: Can we move these up with the other `public` members?

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Grpc/SettlementService.cs [hunk 5]
```diff
@@ -74,16 +73,38 @@ public override async Task<Empty> Deadletter(DeadletterRequest request, ServerCa
...
+        private static Dictionary<string, object> DeserializeAmqpMap(ByteString mapBytes)
+        {
+            if (mapBytes == null || mapBytes == ByteString.Empty)
+            {
+                return null;
+            }
+
+            var bytes = mapBytes.ToByteArray();
+            using ByteBuffer buffer = new ByteBuffer(bytes.Length, false);
+            AmqpBitConverter.WriteBytes(buffer, bytes, 0, bytes.Length);
```
Comment (jsquire) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Grpc/SettlementService.cs:94]: These are from the AMQP transport library?   Should we consider abstracting them in Core.Amqp or are they specific enough to this scenario that there's no value?

**Review by jsquire — APPROVED** [review]

File: sdk/core/Azure.Core.Amqp/src/Shared/AmqpAnnotatedMessageConverter.cs (hunk 3 above)
Comment (JoshLove-msft) [reply sdk/core/Azure.Core.Amqp/src/Shared/AmqpAnnotatedMessageConverter.cs:662]: It will also be needed by the isolated worker extension.

**Pull request merged**